### PR TITLE
Common Methods and Join Edges for a list with more then two elements

### PR DESCRIPTION
**Pull request by MeikeWeiss**
Issue #61 
Issue #78

### CODE REVIEW

File: gap/PolygonalComplexes/navigation.gi [hunk 1]
```diff
@@ -919,3 +919,296 @@ InstallMethod( FacesWithEdgeProperties,
...
+	[IsVEFComplex,IsList],
+	function(complex,edgeList)
+		local vertices,e,comb,vertex;
+		if Length(edgeList)>1 then
+			vertices:=[];
+			comb:=Combinations(edgeList,2);
+			vertices:=CommonVerticesOfEdgesNC(complex,comb[1][1],comb[1][2]);
+			for e in [2..Length(comb)] do
+				vertices:=Intersection(vertices,CommonVerticesOfEdgesNC(complex,comb[e][1],comb[e][2]));
+			od;
```
Comment (markusbaumeister) [inline gap/PolygonalComplexes/navigation.gi:957]: This seems a bit inefficient. Instead of iterating over all pairs of edges, you could just iterate over the edges, i.e.
vertices := VerticesOfEdgeNC(edgeList[1])
for e in [2..Length(edgeList)] do
  vertices := Intersection(vertices, VerticesOfEdgeNC(e);
od

File: gap/PolygonalComplexes/navigation.gi [hunk 2]
```diff
@@ -919,3 +919,296 @@ InstallMethod( FacesWithEdgeProperties,
...
+			return vertices;
+		elif List(edgeList)=1 then
```
Comment (markusbaumeister) [inline gap/PolygonalComplexes/navigation.gi:959]: You probably mean Length instead of List.
Can you test this method with lists of length 0 and 1 to insure against such edge cases?

File: gap/PolygonalComplexes/navigation.gd [hunk 3]
```diff
@@ -1082,4 +1082,210 @@ DeclareOperation("FacesWithEdgeProperties", [IsVEFComplex, IsList] );
...
+#! not check whether the given <A>edge1</A> and <A>edge2</A> is an edge of <A>complex</A>.
+#! 
+#! The method <K>CommonFacesOfEdges</K>(<A>complex</A>, <A>edgeList</A>) returns the
+#! set of all faces that are incident to all the edges in <A>edgeList</A>. The NC-version does
+#! not check whether the given edges in <A>edgeList</A> are an edge of <A>complex</A>.
+#! 
+#! As an example, consider the polygonal complex that was introduced at the
+#! start of section <Ref Sect="Section_Access_BasicAccess"/>:
+#! <Alt Only="TikZ">
+#!   \begin{tikzpicture}[vertexStyle=nolabels, edgeStyle, faceStyle=]
```
Comment (markusbaumeister) [inline gap/PolygonalComplexes/navigation.gd:1235]: There is an "=" after "faceStyle". This is probably the reason why the corresponding image is not found (the images are encoded by a hash of their source code).

**Review by markusbaumeister — CHANGES_REQUESTED** [review]
This is quite a good start :).

However, some improvement can be made:
1) If I call `gap tst/testall.g`, there are several test failures. This should have been noticed by the CI for this pull request, but the CI did not run (for whatever reason). Please correct these.
2) The individual code corrections (optimisation, testing edge cases) should be done for all common methods (since this concerns all of them).

I also feel that "Common methods" is not a good section header - it is usually interpreted as "commonly used methods", which is not your intention. Unfortunately, I do not have a good suggestion here that correctly communicates the intent and preserves the connection to the word "common".

**Comment by MeikeWeiss** [comment]
Thank you for your review. 

I think I changed everything you mentioned but I'm not sure  if the section header is better now.

**Review by markusbaumeister — APPROVED** [review]
Very nice, this is a very good pull request now :+1:.

While still not optimal (one might quibble about the meaning of "elements"), the new section header is much better understandable. Instead of searching forever for the perfect name, settling one a good one is sufficient here. It can always be changed in the future if someone thinks of something better :).
